plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -426,14 +487,14 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
                 return Optional.of(decimalColumnMapping(createDecimalType(precision, max(decimalDigits, 0))));
 
             case Types.CHAR:
-                return Optional.of(defaultCharColumnMapping(typeHandle.getRequiredColumnSize(), false));
+                return Optional.of(mySqlDefaultCharColumnMapping(typeHandle.getRequiredColumnSize(), typeHandle.getCaseSensitivity().orElse(CASE_INSENSITIVE) == CASE_SENSITIVE));
```

Comment:
`typeHandle.getCaseSensitivity().orElse(CASE_INSENSITIVE)`
It's case insensitive by default, not to allow push downs.

In MySql there is connection/session property `collation_connection` but it's not counted, when we deal with text columns:
`For comparisons of strings with column values, collation_connection does not matter because columns have their own collation, which has a higher collation precedence`

Reply:
Is it possible to have a partial pushdown - like in case of case insensitive - MySQL would match them in case insensitive way and Trino could apply case sensitive checks on top of it.